app/views/editor/thang/replaceRgbaWithCustomizableHex.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * This is an internal script that can be used in the thangEditor to normalize
+ * all the Adobe Animate colors in the shapes to hex values that can be customized.
+ */
+
+const colorBuckets = {
+  hairLight: '#752744',
+  hairMid: '#431D32',
+  hairDark: '#330A1D',
+  skinLight: '#DA6D46',
+  skinMid: '#703E31',
+  skinDark: '#58322A',
+  skinAccent: '#3A1E15'
+}
+
+const rgbaRegex = /rgba\((.*?),(.*?),(.*?),(.*?)\)/g
+
+function hexToRgb (hex) {
+  const extractHex = (start, end) => parseInt(hex.slice(start, end), 16)
+  return [extractHex(1, 3), extractHex(3, 5), extractHex(5, 7)]
+}
+
+// Used to check if two colors are similar.
+const isSimilar = (a, b) => Math.abs(a - b) <= 3
+
+/**
+ * Replaces rgba values with the bucketed hex value.
+ * Tries to detect if the value is similar to one of the bucket values.
+ * If so, will round into a bucket value.
+ * @param s text contents of the file
+ */
+function replaceRgbaWithHex (s) {
+  return s.replace(rgbaRegex, (matchedString, r, g, b, _a) => {
+    const r1 = parseInt(r, 10)
+    const g1 = parseInt(g, 10)
+    const b1 = parseInt(b, 10)
+    // const a1: number = parseFloat(a)
```

Comment:
I assume this is useful in cases where you are working with the alpha. So future readers don't assume this is a left over comment that should be removed, it could be useful to add clarification above, like "Uncomment this to use the alpha value", and have it use the `_a` variable.

Reply:
I don't think alpha will need to be used, so I'll remove all references to alpha.
Great comment!